### PR TITLE
Implement changes required to begin execution of e2e tests.

### PR DESCRIPTION
**Pull request by rbair23**
This PR makes it possible to begin execution of BDD e2e tests (test-clients test suites) for the modular code. This PR **does not** start running them in an automated way, and certainly not as a release gate, because no test is known to pass right now. The purpose of this PR is to allow service owners to begin e2e testing so we can begin seeing these tests pass. We will want to add an automated test execution mode soon, but not as a release gate, so we can measure progress.

This PR introduces a new gradle task, `modrun`, which will build and run a modular consensus node. Once it is running, you can navigate to the BDD test suite of your choice, and execute it from within the IDE. You can start the server in debug mode and have access to the debugger. See [How to run modular e2e tests manually](https://www.loom.com/share/a1311f7b013845a793ea388f66dc0ed5?sid=71e58ddf-1caf-4fe6-b189-b2e6f4158bbb)

This PR includes a lot of fixes to `hedera-app` that were necessary to be able to start the server well enough for some queries to start working. However there are still many things that need to be fixed and addressed before tests start passing. Hopefully the `hedera-app` pieces are close for basic tests to pass, but if there are issues, we will find and fix them in follow up PRs.

This PR:
  - Adds `modrun` as a gradle task (build.gradle.kts)
  - Add support for parsing an `AccountID` from the address book memo field, because this is how we record, for a node in the address book, what account ID that node is associated with (see HapiUtils.java).
  - Introduces `NetworkInfo` and changes `NodeInfo`. Previously, `NodeInfo` was both node and network info. I split these into two classes. `NetworkInfo` can be used to get the `NodeInfo` for any node, including the `selfNodeInfo()`.
  - Adds `recordCache()` to the `QueryContext`
  - Massively changes `Hedera`. Many changes to log statements. Lots of code commented out that is not yet used or relies on the mono service. There will be many more changes to this code in follow on PRs. It isn't necessary to dig too deep on this one right now, as it is in a major state of flux.
  - `HederaInjectionComponent` no longer extends from mono's `ServicesApp`. We want to start cutting the ties that bind us to mono service.
  - Add a `Profile` enum for determining DEV, TEST, or PROD profile mode. This has to be out of Config, because this is a private internal API for `hedera-app`.
  - Remove use of `BootstrapProperties` and use `ConfigProviderImpl` consistently
  - An implementation of `Authorizer` that does not use the mono service code anymore
  - Add support for environment and system properties for configuration of the modular consensus node
  - Remove existing Mono implementations of fees and network info and so forth. Remove from `MerkleHederaState` `StateChildrenProvider` as it is no longer used.
  - Configure use of the notification engine within `Hedera` and implement `CurrentPlatformStatus` without using mono
  - Implement `LedgerValidator` without using mono
  - Implement payer verification in `IngestChecker` without mono
  - Fix bugs in file services where `ReadableFileStoreImpl` was looked up by `createStore` instead of `ReadableFileStore`.
  - Create genesis schema for file service with implementations for genesis creation of some special files
  - Created `InProcessIntegrationTestBase` which can start a new server in process in JUnit instead of using test containers. I'm not sure if this will be generally useful, because creating a docker container is now much faster than it was (in #7264) and there is currently no good way to run a single test at a time for dynamically created tests. So maybe this one will not last long. We'll see.

### CODE REVIEW

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java [hunk 1]
```diff
@@ -237,32 +257,36 @@ public static Stream<Arguments> failureReasons() {
...
-            doThrow(new RuntimeException("checkPayerSignature exception"))
-                    .when(signaturePreparer)
-                    .syncGetPayerSigStatus(any());
-
-            // When the transaction is submitted, then the exception is bubbled up
-            assertThatThrownBy(() -> subject.runAllChecks(state, tx))
-                    .isInstanceOf(RuntimeException.class)
-                    .hasMessageContaining("checkPayerSignature exception");
-        }
+        //        @ParameterizedTest(name = "SignatureCheck fails with error code {0}")
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java:260]: Should these be uncommented ?

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM !

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 2]
```diff
@@ -617,30 +802,36 @@ private void restart(
...
+        //        final var stakingInfo = stateChildren.stakingInfo();
+        //        final var networkCtx = stateChildren.networkCtx();
+        //        daggerApp.stakeStartupHelper().doRestartHousekeeping(stateChildren.addressBook(), stakingInfo);
+        //        if (upgrade) {
+        //            dualState.setFreezeTime(null);
+        //            networkCtx.discardPreparedUpgradeMeta();
+        ////            if (version.hasMigrationRecordsFrom(deserializedVersion)) {
+        ////                networkCtx.markMigrationRecordsNotYetStreamed();
+        ////            }
+        //        }
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:824]: Multiple comments on this file will go into the methods implementations ?

File: hedera-node/test-clients/src/itest/java/InProcessIntegrationTestBase.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        registry.registerConstructables("com.swirlds.merkle");
+        registry.registerConstructables("com.swirlds.merkle.tree");
+
+        // 1. Create a configuration instance with any desired overrides.
+        final var factory = ServiceLoader.load(ConfigurationBuilderFactory.class);
+        final var configBuilder = factory.findFirst().orElseThrow().create();
+        final var config = configBuilder
+                //                .withSource(mappedSettingsConfigSource)
+                //                .withSource(configPropertiesConfigSource)
+                //                .withSource(threadCountPropertyConfigSource)
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/itest/java/InProcessIntegrationTestBase.java:113]: Intentionally commented out?

Comment (rbair23) [reply hedera-node/test-clients/src/itest/java/InProcessIntegrationTestBase.java:113]: Yes, it is a work in progress. But I can remove these.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java (hunk 2 above)
Comment (rbair23) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:824]: I'm not entirely sure. I'll work on getting these lines replaced and done right, but at the moment we don't need them to at least start doing e2e testing for basic things

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java (hunk 1 above)
Comment (rbair23) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java:260]: Oh thanks! I will get those working again.

**Review by nathanklick — APPROVED** [review]
Limited approval only for the `module-info.java` and `*.kts` file changes.

**Review by Neeharika-Sompalli — DISMISSED** [review]

**Review by Neeharika-Sompalli — COMMENTED** [review]
It has some conflicts to be resolved though

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/LedgerValidator.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.state;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public interface LedgerValidator {
```
Comment (netopyr) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/LedgerValidator.java:21]: JavaDocs

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java [hunk 5]
```diff
@@ -123,15 +125,15 @@ public class TransactionChecker {
     public TransactionChecker(
             @MaxSignedTxnSize final int maxSignedTxnSize,
             @NodeSelfId @NonNull final AccountID nodeAccount,
-            @NonNull final GlobalDynamicProperties dynamicProperties,
+            @NonNull final ConfigProvider configProvider,
```
Comment (netopyr) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java:128]: I think it is cleaner, if we pass the configuration as a parameter. Can be done in a separate PR.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java [hunk 6]
```diff
@@ -104,21 +123,83 @@ public TransactionInfo runAllChecks(@NonNull final HederaState state, @NonNull f
...
+        // Verify the signatures
+        final var results = signatureVerifier.verify(txInfo.signedBytes(), expandedSigs);
+        final var future = results.get(payerAccount.keyOrThrow());
+
+        // This can happen if the signature map was missing a signature for the payer account.
+        if (future == null) {
+            throw new PreCheckException(INVALID_SIGNATURE);
+        }
+
+        // Wait for the verification to complete. We have a timeout here of 1 second, which is WAY more time
```
Comment (netopyr) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java:181]: This is done in a background thread. Does it matter if we wait 1 or 10 seconds? GC pauses are rare but happen.

File: hedera-node/hedera-config/src/test/java/com/hedera/node/config/PropertySourceBasedConfigTest.java [hunk 7]
```diff
@@ -133,7 +130,7 @@ void testConfig() {
         final FileID fileID = configuration.getValue("test.fileID", FileID.class);
         final HederaFunctionality hederaFunctionality =
                 configuration.getValue("test.hederaFunctionality", HederaFunctionality.class);
-        final Profile profile = configuration.getValue("test.profile", Profile.class);
```
Comment (netopyr) [inline hedera-node/hedera-config/src/test/java/com/hedera/node/config/PropertySourceBasedConfigTest.java:136]: Why did you switch to a `String`?

**Review by netopyr — DISMISSED** [review]
LGTM (at least the parts I am familiar with) 🙂

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenServiceImpl.java [hunk 8]
```diff
@@ -85,18 +86,29 @@ public void migrate(@NonNull MigrationContext ctx) {
...
                 }
                 final var superUserKey =
                         Key.newBuilder().ed25519(superUserKeyBytes).build();
 
+                long remainingBalance = ledgerConfig.totalTinyBarFloat();
                 try {
+                    accounts.put(
+                            AccountID.newBuilder().accountNum(3).build(),
+                            Account.newBuilder()
+                                    .accountNumber(2)
```
Comment (iwsimon) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenServiceImpl.java:102]: .accountNumber(3) here?

**Review by iwsimon — APPROVED** [review]
One comment to check, LGTM

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]
Looks Great ! Thanks @rbair23 🙌

**Review by kimbor — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

**Review by jasperpotts — APPROVED** [review]

**Pull request merged**